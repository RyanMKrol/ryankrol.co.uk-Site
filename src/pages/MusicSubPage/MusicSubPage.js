import React, { Component } from 'react';

import { DataCollection, ListensItemMini } from '../../components';

import { baseEndpoint } from '../../utils/ProdUtils';

/**
 * Page for rendering sub-section of Music content
 */
class MusicSubPage extends Component {
  /**
   * @class
   */
  constructor() {
    super();

    this.endpoint = `${baseEndpoint()}/listens`;
    this.outputLimit = undefined;
  }

  /**
   * Render method
   *
   * @returns {string} JSX Content
   */
  render() {
    return (
      <div>
        <h2>Tune in to Ryan.FM</h2>
        <p>
          I spend a lot of time cultivating my interest in music, finding new artists, and expanding
          my tastes. You can find my account on
          <a href="https://www.last.fm/user/somethingmeaty">
            <span className="highlighted"> LastFM </span>
          </a>
          where I'm always tracking what I'm listening to. The albums below show the albums that
          I've listened to the most over the last year (don't judge!).
        </p>
        <p>
          Note: Sadly I'm no longer able to track this data as I've moved to Apple Music, which
          doesn't appear to support tracking data via LastFM. I still quite like the data I've
          amassed over the past couple of years though, so I'll keep this section for now.
        </p>
        <DataCollection
          outputSize={this.outputLimit}
          endpoint={this.endpoint}
          itemTag={ListensItemMini}
        />
      </div>
    );
  }
}

export default MusicSubPage;
