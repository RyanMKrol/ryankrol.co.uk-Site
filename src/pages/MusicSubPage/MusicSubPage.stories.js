import React from 'react';

import MusicSubPage from './MusicSubPage';

export default {
  title: 'Pages/MusicSubPage',
};

/**
 * default story
 *
 * @returns {React.Component} JSX for the MusicSubPage
 */
export const normal = () => <MusicSubPage />;
