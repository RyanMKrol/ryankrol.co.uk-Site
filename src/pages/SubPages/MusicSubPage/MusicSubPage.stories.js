import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import MusicSubPage from './MusicSubPage';

export default {
  title: 'Pages/MusicSubPage',
};

/**
 * default story
 *
 * @returns {React.Component} JSX for the MusicSubPage
 */
export const normal = () => (
  <Router>
    <MusicSubPage />
  </Router>
);
