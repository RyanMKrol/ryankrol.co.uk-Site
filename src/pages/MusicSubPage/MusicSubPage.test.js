import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import MusicSubPage from './MusicSubPage';

it('renders playground correctly', () => {
  const header = shallow(<MusicSubPage />);
  expect(toJson(header)).toMatchSnapshot();
});

it('renders playground without throwing', () => {
  expect(() => mount(<MusicSubPage />)).not.toThrow();
});
