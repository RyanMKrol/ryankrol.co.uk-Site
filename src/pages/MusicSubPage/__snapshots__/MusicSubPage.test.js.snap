// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders playground correctly 1`] = `
<div>
  <h2>
    Tune in to Ryan.FM
  </h2>
  <p>
    I spend a lot of time cultivating my interest in music, finding new artists, and expanding my tastes. You can find my account on
    <a
      href="https://www.last.fm/user/somethingmeaty"
    >
      <span
        className="highlighted"
      >
         LastFM 
      </span>
    </a>
    where I'm always tracking what I'm listening to. The albums below show the albums that I've listened to the most over the last year (don't judge!).
  </p>
  <p>
    Note: Sadly I'm no longer able to track this data as I've moved to Apple Music, which doesn't appear to support tracking data via LastFM. I still quite like the data I've amassed over the past couple of years though, so I'll keep this section for now.
  </p>
  <DataCollection
    endpoint="http://www.ryankrol.co.uk/api/listens"
    itemTag={[Function]}
  />
</div>
`;
